gdpr/models.py

```diff
@@ -11,7 +12,64 @@
 
 class LegalReasonManager(models.Manager):
 
-    def create_from_purpose_slug(self, purpose_slug, source_object, issued_at=None, tag=None, related_objects=None):
+    def create_consent(self, purpose_slug, source_object, issued_at=None, tag=None, related_objects=None):
+        """
+        Create (or update, if it exist) a LegalReason with purpose slug for concrete object instance
+
+        Args:
+            purpose_slug: String of Legal Reason purpose
+            source_object: Source object this Legal Reason is related to
+            issued_at: When the Legal Reason consent was given
+            tag:
+            related_objects: Objects this Legal Reason relates to (ie. order, registrations etc.)
+
+        Returns:
+            Legal Reason: LegalReason object
+        """
+        if self.filter_source_instance(source_object).filter(purpose_slug=purpose_slug).exists():
+            return self._refresh_for_source_by_slug(purpose_slug, source_object, tag, related_objects)
+        else:
+            return self._create_from_purpose_slug(purpose_slug, source_object, issued_at, tag, related_objects)
+
+    def deactivate_consent(self, purpose_slug, source_object):
+        """
+        Deactivate/Remove consent (Leagal reason) for source_object, purpose_slug combination
+
+        Args:
+            purpose_slug: Purpose slug to deactivate consent for
+            source_object: Source object to deactivate consent for
+        """
+        LegalReason.objects.filter_source_instance_active_non_expired(source_object).filter(
+            purpose_slug=purpose_slug).update(is_active=False)
```

Comment:
teoreticky by měl existovat jen jeden. Takže toto by šlo dělat jednodušeji

Reply:
ale klidně to budeme refaktorovat později

Reply:
jo a přidat testy taky... (později)

Reply:
To už je zapsáno v těch vylepšních, která by se měla udělat.